pkg/sql/alter_table.go

```diff
@@ -810,29 +749,31 @@ func (n *alterTableNode) startExec(params runParams) error {
 			descriptorChanged = descriptorChanged || descChanged
 
 		case *tree.AlterTableRenameConstraint:
-			info, err := n.tableDesc.GetConstraintInfo()
-			if err != nil {
-				return err
-			}
-			details, ok := info[string(t.Constraint)]
-			if !ok {
+			constraint, _ := n.tableDesc.FindConstraintWithName(string(t.Constraint))
+			if constraint == nil {
```

Comment:
So many calls to `GetConstraintInfo` were really about finding a constraint by name. Though, notably, `GetConstraintInfo` would not return the full set of constraints (no dropping unique indexes for instance). In this case here I don't think that mattered but I'm willing to bet this caused some subtle undetected bugs at some other `GetConstraintInfo` call sites.